doc/src/devdocs/gc.md

```diff
@@ -1,61 +1,74 @@
-  from `page_pool_freed`, which contains pages which have already been madvised by a concurrent sweeper GC thread and whose underlying virtual address can be recycled.
+- Try to claim a block from `page_pool_lazily_freed`, which contains blocks that were empty during the last stop-the-world phase, but haven't been madvised by a concurrent scavenger GC thread yet.
 
-- If it failed in all of the attempts mentioned above, it will mmap a batch of pages, claim one page for itself, and
-  insert the remaining pages into `page_pool_clean`.
+- If it failed to claim a block from `page_pool_lazily_freed`, it will try to claim a block from `page_pool_clean`, which contains blocks mapped on a previous block allocation request but never accessed.
+
+- If it failed to claim a block from `page_pool_clean` and from `page_pool_lazily_freed`, it will try to claim a block from `page_pool_freed`, which contains blocks already scavenged by a concurrent scavenger GC thread and whose underlying virtual address can be recycled.
+
+- If it failed in all of the attempts mentioned above, it will map a batch of operating system pages, partition them into memory blocks, claim one block for itself, and insert the remaining blocks into `page_pool_clean`.
 
 ![Diagram of tiered pool allocation](./img/gc-tiered-allocation.jpg)
 
+### Large Object Allocation
+
+Sufficiently large objects, above the 2k byte threshold mentioned in the previous section, are allocated through `libc` `malloc`. Large allocations are typically less performance-critical than small allocations, as they occur less frequently.
+
+Although Julia currently uses `libc` `malloc`, it also supports pre-loading other dynamic memory allocators (e.g., `jemalloc`).
+
 ## Marking and Generational Collection
 
-Julia’s mark phase is implemented through a parallel iterative depth-first-search over the object graph. Julia’s collector is non-moving, so object age information can’t be determined through the memory region in which the object resides alone, but has to be somehow encoded in the object header or on a side table. The lowest two bits of an object’s header are used to store, respectively, a mark bit that is set when an object is scanned during the mark phase and an age bit for the generational collection.
+Julia’s mark phase is implemented through a parallel depth-first-search that traverses the object graph to determine which objects are alive.
+
+Because Julia’s GC is non-moving, object age information can’t be only determined through the object's memory address, such as in GC implementations that allocate young objects in certain memory regions and relocate them to other memory regions during object promotion. Age information needs to be encoded in the object header or on a side table. Julia implements the former: the lowest two bits of an object’s header store a mark bit, set when an object is marked, and an age bit, set when the object is promoted.
```

Comment:
I don't think the comparison to relocating here is necessary and adds some confusion.

It would be clearer to first introduce the fact that Julia allocates a header with the type tag ahead of the object and that we use bits in those for marking and for generation.